### PR TITLE
rust-analyzer 07-06

### PR DESCRIPTION
**Pull request by englishm**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by englishm** [comment]
This appears to the be the failure:

```json
{
  "jsonrpc": "2.0",
  "method": "window/showMessage",
  "params": {
    "message": "rust-analyzer failed to discover workspace",
    "type": 1
  }
}
```

https://github.com/Homebrew/homebrew-core/pull/57582/checks?check_run_id=843642664#step:6:68

It looks like the test for this formula does some RPC calls, but doesn't actually set up a `Cargo.toml`. 
Something may have changed since the last version to break the assumptions this test rests on.

File: Formula/rust-analyzer.rb [hunk 1]
```diff
@@ -51,7 +51,7 @@ def rpc(json)
       "jsonrpc":"2.0",
       "id": 1,
       "method":"shutdown",
-      "params": {}
+      "params": null
```
Comment (englishm) [inline Formula/rust-analyzer.rb:54]: In some older versions of the LSP spec, the `shutdown` request was specified as:

> method: 'shutdown'
> params: undefined

This appears to be the case as recently as [version 3.0](https://github.com/microsoft/language-server-protocol/blob/164eb31bd49535dca046034602146f96bb17b56d/protocol.md#shutdown). However, sometime between version 3.0 and [version 3.14](https://github.com/microsoft/language-server-protocol/blob/04f1091f353f1c657f6379002ff40b3f886b0fd9/_specifications/specification-3-14.md#shutdown), of the LSP specification, the shutdown request was changed to be defined as:

> method: ‘shutdown’
> params: void

In JSON, these values are typically mapped to `{}` and `null`, respectively.

I haven't found exactly where in the history of `rust-analyzer` or `lsp-server` this field may have started being parsed more strictly in conformance with recent LSP specs, but this minor change appears to be all that's needed to get the tests passing again.

**Comment by chenrui333** [comment]
@englishm thanks for the PR, can you squash your commits into `rust-analyzer 07-06`?

**Comment by englishm** [comment]
Thanks for looking at this @chenrui333!

I've squashed dd5a4796e9 and bc8b03b2f1 into d751172da1.

**Comment by chenrui333** [comment]
CI hiccup, need to rebuild again.

**Review by SMillerDev — APPROVED** [review]
Thanks @englishm  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**